rpc: deprecate top-level fee fields in getmempool RPCs (#22689)
bitcoin/bitcoin: doc/release-notes.md

Updated RPCs
------------
- The top-level fee fields `fee`, `modifiedfee`, `ancestorfees` and `descendantfees`
  returned by RPCs `getmempoolentry`,`getrawmempool(verbose=true)`,
  `getmempoolancestors(verbose=true)` and `getmempooldescendants(verbose=true)`
  are now deprecated and will be removed in the next major version (use
  `-deprecated=fees` if needed in this version). The same values, denominated

>   in BTC, can be accessed through the `fees` object in the result. (#22689)

Reviewer: The units don't seem to be changed and remain in BTC/kvB?

Feel free to ignore (the help prints the wrong unit as well, I believe)
```suggestion
  in BTC/kvB, can be accessed through the `fees` object in the result. (#22689)
```

Author: yeah, i think the help is incorrect here. i can open a general RPC cleanup PR. for starters:

- address where argument order doesn't match help order
- fix fee vs feerate and other small errors

Reviewer: So, `fee` and `modifiedfee` are uniformly in BTC/kvB. OTOH, it looks to me like `ancestorfees` and `descendantfees` are in sat/vB in the deprecated fields and in BTC/kvB in the `fees` object (urgh, we've been moving from BTC/kvB to sat/vB elsewhere).

If we mention this in the release notes (and I think it's a good idea as @glozow suggested), it may be best to (a) use the correct units of BTC/kvB and (b) clarify which fields change from sat/vB to BTC/kvB (`ancestorfees` and `descendantfees`) in the release notes.

Finally, here's a commit in https://github.com/jonatack/bitcoin/commits/pr22689-help-fixups that both fixes and improves the docs for these fields, feel free to use or pull it in here since these lines are already being touched (or later, if you prefer) as the helps could use some...help :)

Reviewer 2: Note that the units for ancestor and descendant fees are different - they were in satoshis and are now in BTC

Updated RPCs
------------
- The top-level fee fields `fee`, `modifiedfee`, `ancestorfees` and `descendantfees`
  returned by RPCs `getmempoolentry`,`getrawmempool(verbose=true)`,
  `getmempoolancestors(verbose=true)` and `getmempooldescendants(verbose=true)`
  are now deprecated and will be removed in the next major version (use
  `-deprecated=fees` if needed in this version). The same fee rates, all denominated
  in BTC/kvB, can be accessed through the `fees` object in the result. Note: deprecated
  fields `ancestorfees` and `descendantfees` are denominated in sat/vB, whereas all

>   fields in the `fees` object are denomainted in BTC/kvB (#22689)

Reviewer: ```suggestion
  fields in the `fees` object are denominated in BTC/kvB (#22689)
```

perhaps `s/Note:/Warning:`

Author: oof, good catch. i need to enable spell-check for vim :roll_eyes:

Updated RPCs
------------
- RPCs `getmempoolentry`, `getrawmempool verbose=true`, `getmempoolancestors
  verbose=true` and `getmempooldescendants verbose=true` no longer return the
  top-level fee fields `fee`, `modifiedfee`, `ancestorfees` and
  `descendantfees`. These fields were deprecated in v0.17 and moved to the
  `fees` JSON object in the response. The `-deprecated=fees` configuration
  option must be passed for these fields to be included in the top-level

>   response. This option will only be available until v24.0, after which the

Reviewer 3: nit: It won't be available in v24.0

```suggestion
  response. This option will only be available in this major version, after which the
```

> - RPCs `getmempoolentry`, `getrawmempool verbose=true`, `getmempoolancestors
>   verbose=true` and `getmempooldescendants verbose=true` no longer return the
>   top-level fee fields `fee`, `modifiedfee`, `ancestorfees` and
>   `descendantfees`. These fields were deprecated in v0.17 and moved to the
>   `fees` JSON object in the response. The `-deprecated=fees` configuration
>   option must be passed for these fields to be included in the top-level
>   response. This option will only be available in this major version, after
>   which the fields will be fully removed. Use the `fees` JSON object for these
>   RPCS instead. (#22689)

Reviewer 2: Mostly wording suggestions, but also I think you should warn users about the fact that the fees are denominated slightly differently when they switch to using the `fees` object.
```suggestion
- The top-level fee fields `fee`, `modifiedfee`, `ancestorfees` and `descendantfees`
  returned by RPCs `getmempoolentry`,`getrawmempool(verbose=true)`,
  `getmempoolancestors(verbose=true)` and `getmempooldescendants(verbose=true)`
  are now deprecated and will be removed in the next major version (use
  `-deprecated=fees` if needed in this version). The same values, denominated
  in satoshis, can be accessed through the `fees` object in the result. (#22689)
```

Reviewer 2: Doesn't seem to be an issue here, but for future reference, you might want to put it in a release-notes-22689.md to avoid conflicts with other PRs updating release notes.

Author: > slightly differently when they switch to using the fees object

just double checking, the update should read "denominated in BTC," correct?

Reviewer 2: yes, BTC, sorry

Updated RPCs
------------
- `getmempoolentry` and `getrawmempool true` no longer return top level fee fields `fee`, `modifiedfee`, `ancestorfees`, `descendantfees`. These fields were deprecated in `0.17` and moved to the `fees` sub-object in the response.

> The `-deprecated=fees` flag must be passed for these fields to be included in the top level response. Note: this flag will only be available until `v24`, after which the fields will be fully removed. It is recommended you migrate to using the `fees` sub-object in the response

Reviewer: suggestions

- add the PR number
- add  `getmempoolancestors` and `getmempooldescendants` that are also changed
- s/true/verbose=true/ for clarity
- wrap the lines
- s/sub-object/JSON object/
- s/top level/top-level/
- a few others

```diff
-- `getmempoolentry` and `getrawmempool true` no longer return top level fee fields `fee`, `modifiedfee`, `ancestorfees`, `descendantfees`. These fields were deprecated in `0.17` and moved to the `fees` sub-object in the response.
-The `-deprecated=fees` flag must be passed for these fields to be included in the top level response. Note: this flag will only be available until `v24`, after which the fields will be fully removed. It is recommended you migrate to using the `fees` sub-object in the response
+- RPCs `getmempoolentry`, `getrawmempool verbose=true`, `getmempoolancestors
+  verbose=true` and `getmempooldescendants verbose=true` no longer return the
+  top-level fee fields `fee`, `modifiedfee`, `ancestorfees` and
+  `descendantfees`. These fields were deprecated in v0.17 and moved to the
+  `fees` JSON object in the response. The `-deprecated=fees` configuration
+  option must be passed for these fields to be included in the top-level
+  response. This option will only be available until v24.0, after which the
+  fields will be fully removed. Use the `fees` JSON object for these RPCS
+  instead. (#22689)
```

Updated RPCs
------------

- The top-level fee fields `fee`, `modifiedfee`, `ancestorfees` and `descendantfees`
  returned by RPCs `getmempoolentry`,`getrawmempool(verbose=true)`,
  `getmempoolancestors(verbose=true)` and `getmempooldescendants(verbose=true)`
  are now deprecated and will be removed in the next major version (use

>   `-deprecated=fees` if needed in this version). The same fee rates, all denominated
>   in BTC/kvB, can be accessed through the `fees` object in the result. Warning: deprecated

Reviewer 2: ```suggestion
  `-deprecated=fees` if needed in this version). The same fee, all denominated
  in BTC, can be accessed through the `fees` object in the result. Warning: deprecated
```
